juju/juju
# [JUJU-796] Remove some alias commands

Here is a full list of the command changes from `juju`:
- Drop `upgrade-charm` in favour of `refresh`
- Drop `remove-consumed-application` in favour of `remove-saas`
- Drop `hook-tool` and `hook-tools` in favour of `help-tool`
- Drop `sync-tools` in favour of `sync-agent-binary`
- Drop `upgrade-juju` in favour of `upgrade-model`
- Drop `attach` in favour of `attach-resource`
- Drop `show-status` in favour of `status`

And from `juju-metadata`
- Drop `generate-tools`/`generate-agents` and change to `generate-agent-binaries`
- Drop `validate-tools`/`validate-agents` and change to `validate-agent-binaries`
- Add `images` to `list-images` for consistency

Also, consistently make list-* variants aliases where applicable, and drop the `list-` alias from examples

## Checklist


## QA steps

```sh
make static-analysis
go test github.com/juju/juju/cmd/juju/...
go test github.com/juju/juju/cmd/plugins/juju-metadata
```

```sh
$ juju show-status
ERROR juju: "show-status" is not a juju command. See "juju --help".
```

## Documentation changes

Probably does require a documentation change or two

## Bug reference

N/A
